### DISCUSSION TITLE
citation autocomplete?

### DISCUSSION THREAD

**Discussion by Lightup1** [quarto-dev/quarto-cli#1639] [Q&A] [answered]
Is there a way to autocomplete citation as overleaf?
Or connect to Zotero or Mendeley?

**Comment by Lightup1** [comment]
Never mind. Since Zotero has vscode extension supporting CAYW, this issue should be resolved for Zotero users.
But for quarto extension, it will be good if cross ref and citation can all be autocompleted.

**Comment by tbeason** [comment]
I would also like this functionality! VS Code LaTeX Workshop extension has this and I did not realize how helpful it was now that I'm without it.

**Comment by jjallaire** [comment, marked as answer]
We actually just released support for this yesterday: https://twitter.com/quarto_pub/status/1554538304367902720

**Comment by dkgaraujo** [comment]
Not sure if an issue, but I am using Quarto with VS Code and citation autocomplete has always worked well... until it didn't. No visible configuration change from my side; nothing can get it to return to the same (very helpful) autocomplete for citations as existed before. Not even disabling and re-enabling the Quarto extension. Maybe this experience is helpful for devs. If you would like me to raise an issue, I could add more formal information.

**Reply by ghost** [reply]
Thanks for posting this.  Yes, [please file a bug](https://github.com/quarto-dev/quarto-cli/issues/new?assignees=&labels=bug&template=bug_report.yml) and we will take a look.  If we have introduced a bug, we want to fix it.

**Reply by DedZago** [reply]
Hi, I also want to point out that Quarto citation autocomplete in VS Code has stopped working.
Furthermore, I have not found a way to disable this citation autocomplete feature, and the pandoc underlying process runs indefinitely consuming a high amount of RAM and CPU. The only way for me to fix this is to kill the pandoc process every time I (accidentally) ask for citation autocompletion.

**Reply by mcanouil** [reply]
Please file a bug report (if none already exists) on <https://github.com/quarto-dev/quarto>.
Note that you need to provide more information than only "Quarto citation autocomplete in VS Code has stopped working" which does not give context such as VSCode version, VSCode Quarto extension version, the step to reproduce (both the "memory issue" and autocomplete).

For instance, on macOS with the latest version of everything, I don't see any issues.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/ded21a65-dfa3-45e4-a423-c6a54c9f7cb4">

**Reply by cderv** [reply]
I already moved this to 
- https://github.com/quarto-dev/quarto/issues/410

We can keep discussing there https://github.com/quarto-dev/quarto/issues/410
